src/views/thread/components/messages.js

```diff
@@ -147,33 +146,55 @@ class MessagesWithData extends React.Component<Props, State> {
       isLoading,
       toggleReaction,
       forceScrollToBottom,
-      hasMessagesToLoad,
       id,
-      isModerator,
       isFetchingMore,
       loadPreviousPage,
       loadNextPage,
       scrollContainer,
       location,
-      threadIsLocked,
       lastSeen,
+      thread,
+      currentUser,
     } = this.props;
 
-    const dataExists =
+    const hasMessagesToLoad = thread.messageCount > 0;
+    const { channelPermissions } = thread.channel;
+    const { communityPermissions } = thread.community;
+    const { isLocked } = thread;
+    const isChannelOwner = currentUser && channelPermissions.isOwner;
+    const isCommunityOwner = currentUser && communityPermissions.isOwner;
+    const isChannelModerator = currentUser && channelPermissions.isModerator;
+    const isCommunityModerator =
+      currentUser && communityPermissions.isModerator;
+    const isModerator =
+      isChannelOwner ||
+      isCommunityOwner ||
+      isChannelModerator ||
+      isCommunityModerator;
+
+    const messagesExist =
       data &&
       data.thread &&
       data.thread.id === id &&
-      data.thread.messageConnection;
-    const messagesExist =
-      dataExists && data.thread.messageConnection.edges.length > 0;
+      data.thread.messageConnection &&
+      data.thread.messageConnection.edges.length > 0;
 
     if (messagesExist) {
       const { edges, pageInfo } = data.thread.messageConnection;
-      const unsortedMessages = edges.map(message => message.node);
+      const unsortedMessages = edges.map(message => message && message.node);
 
       const uniqueMessages = deduplicateChildren(unsortedMessages, 'id');
       const sortedMessages = sortAndGroupMessages(uniqueMessages);
 
+      const prevCursor =
+        edges && edges.length > 0 && edges[0] && edges[0].cursor;
+      const lastEdge =
+        edges &&
+        edges.length > 0 &&
+        edges[edges.length - 1] !== (null || undefined) &&
```

Comment:
This don't work, `(null || undefined)` evaluates to false. This check is unnecessary anyway, just make it return `edges[edges.length - 1]`!